rtnl/link.go

```diff
@@ -0,0 +1,75 @@
+// github.com/jsimonetti/rtnetlink/rtnl
+package rtnl
+
+import (
+	"github.com/jsimonetti/rtnetlink"
+	"golang.org/x/sys/unix"
+	"net"
+)
+
+func Links(conn *rtnetlink.Conn) ([]rtnetlink.LinkMessage, error) {
+	msg, err := conn.Link.List()
+	if err != nil {
+		return nil, err
+	}
+	return msg, nil
+}
+
+func LinkSetHardwareAddr(conn *rtnetlink.Conn, ifc *net.Interface, hw net.HardwareAddr) error {
+	msg, err := conn.Link.Get(uint32(ifc.Index))
+	if err != nil {
+		return err
+	}
+	tx := &rtnetlink.LinkMessage{
+		Family: msg.Family,
+		Type:   msg.Type,
+		Index:  uint32(ifc.Index),
+		Flags:  msg.Flags,
+		Change: msg.Change,
+		Attributes: &rtnetlink.LinkAttributes{
+			Address: hw,
+		},
+	}
+	err = conn.Link.Set(tx)
+	return err
+}
+
+func LinkUp(conn *rtnetlink.Conn, ifc *net.Interface) (err error) {
+	msg, err := conn.Link.Get(uint32(ifc.Index))
+	if err != nil {
+		return err
+	}
+	state := msg.Attributes.OperationalState
+	if state == rtnetlink.OperStateUp || state == rtnetlink.OperStateUnknown {
```

Comment:
I think this logic should be in the caller (e.g. see https://github.com/insomniacslk/dhcp/blob/master/netboot/netconf.go#L150 , where the caller also explains *why* those two states).

This (having to take a decision based on the state) is also another indicator that Get before Set may not be the right thing to do

Reply:
Agreed.